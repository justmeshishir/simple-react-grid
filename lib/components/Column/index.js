import React from 'react';
import styled from 'styled-components';
import PropTypes from 'prop-types';

let getWidthStyle = (size) => {
  if (!size) return;
  return `width: ${ size / 12 * 100 }%;`;
};

const Column = styled.div`
  float: left;

  ${({ sm }) => (sm ? getWidthStyle(sm): 'width: 100%')};

  @media only screen and (min-width: 768px) {
    ${({ md }) => md && getWidthStyle(md)};
  }

  @media only screen and (min-width: 992px) {
    ${({ lg }) => lg && getWidthStyle(lg)};
  }
`;

Column.propTypes = {
  lg: PropTypes.string,
  md: PropTypes.string,
  sm: PropTypes.string,
};

export default Column;
