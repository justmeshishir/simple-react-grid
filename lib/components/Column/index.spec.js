import React from 'react';
import renderer from 'react-test-renderer';

import Column from './index';

describe('snapshot', () => {
  it('renders correctly with defaults', () => {
    const tree = renderer.create(
      <Column />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with sm="6"', () => {
    const tree = renderer.create(
      <Column sm="6" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with md="6"', () => {
    const tree = renderer.create(
      <Column md="6" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with lg="6"', () => {
    const tree = renderer.create(
      <Column lg="6" />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
