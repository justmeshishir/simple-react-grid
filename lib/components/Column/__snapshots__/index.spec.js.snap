// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot renders correctly with defaults 1`] = `
<div
  className="sc-bdVaJa kHaRMQ"
/>
`;

exports[`snapshot renders correctly with lg="6" 1`] = `
<div
  className="sc-bdVaJa fDOotm"
/>
`;

exports[`snapshot renders correctly with md="6" 1`] = `
<div
  className="sc-bdVaJa epnEPF"
/>
`;

exports[`snapshot renders correctly with sm="6" 1`] = `
<div
  className="sc-bdVaJa liwDMx"
/>
`;
